packages/flame/lib/src/geometry/polygon.dart

```diff
@@ -51,17 +56,29 @@ class Polygon extends Shape {
     Vector2? position,
     Vector2? size,
     double? angle,
-  }) : super(position: position, size: size, angle: angle ?? 0);
+  }) : super(position: position, size: size, angle: angle ?? 0) {
+    _sizedVertices = List.generate(
+      normalizedVertices.length,
+      (_) => Vector2.zero(),
+      growable: false,
+    );
+    _hitboxVertices = List.generate(
```

Comment:
I am not completely sure, but I think List.filled would be better here

Reply:
List.filled would generated a list with only the same vector on all positions, I need new objects on each position as I use these as a type of static vector pool.

Reply:
Makes sense!

Reply:
could this just be `normalizedVertices.map((_) => Vector2.zero())`?

Reply:
Ah, much nicer, changed to that!